feature_engine/selection/recursive_feature_addition.py

```diff
@@ -0,0 +1,269 @@
+        """
+
+        Args
+        ----
+
+        X: pandas dataframe of shape = [n_samples, n_features]
+           The input dataframe
+
+        y: array-like of shape (n_samples)
+           Target variable. Required to train the estimator.
+
+
+        Returns
+        -------
+
+        self
+        """
+
+        # check input dataframe
+        X = _is_dataframe(X)
+
+        # find numerical variables or check variables entered by user
+        self.variables = _find_or_check_numerical_variables(X, self.variables)
+
+        # train model with all features and cross-validation
+        model = cross_validate(
+            self.estimator,
+            X[self.variables],
+            y,
+            cv=self.cv,
+            scoring=self.scoring,
+            return_estimator=True,
+        )
+
+        # store initial model performance
+        self.initial_model_performance_ = model["test_score"].mean()
+
+        # Initialize a dataframe that will contain the list of the feature/coeff
+        # importance for each cross validation fold
+        feature_importances_cv = pd.DataFrame()
+
+        # Populate the feature_importances_cv dataframe with columns containing
+        # the feature importance values for each model returned by the cross
+        # validation.
+        # There are as many columns as folds.
+        for m in model["estimator"]:
+
+            feature_importances_cv[m] = get_feature_importances(m)
+
+        # Add the variables as index to feature_importances_cv
+        feature_importances_cv.index = self.variables
+
+        # Aggregate the feature importance returned in each fold
+        self.feature_importances_ = feature_importances_cv.mean(axis=1)
+
+        # Sort the feature importance values descreasingly
+        self.feature_importances_.sort_values(ascending=False, inplace=True)
+
+        # Extract most important feature from the ordered list of features
+        first_most_important_feature = list(self.feature_importances_.index)[0]
+
+        # Run baseline model using only the most important feature
+        baseline_model = cross_validate(
+            self.estimator,
+            X[first_most_important_feature].to_frame(),
+            y,
+            cv=self.cv,
+            scoring=self.scoring,
+            return_estimator=True,
+        )
+
+        # Save baseline model performance
+        baseline_model_performance = baseline_model["test_score"].mean()
+
+        # list to collect selected features
+        # It is initialized with the most important feature
+        self.selected_features_ = [first_most_important_feature]
+
+        # dict to collect features and their performance_drift
+        # It is initialized with the performance drift of
+        # the most important feature
+        self.performance_drifts_ = {
+            first_most_important_feature: baseline_model_performance -
+            self.initial_model_performance_
+        }
```

Comment:
I think now, because we are  adding features, we are measuring increase in performance respect to the model with just 1 feature first, and then to the model with additional features, as we add them through.

So for feature 1, the performance drift should be 0 or nan.

Or are you thinking of determining the drift respect to the original model with all features?